README.md

```diff
@@ -107,6 +107,8 @@ r.PathPrefix("/products/")
 
 ```go
 r.Methods("GET", "POST")
+// This is equivalent to .Methods("GET")
```

Comment:
This looks incorrect to me :(

Because when we write multiple `Methods` calls like: `Methods("GET", "POST").Methods("PATCH")`

Apparently, In one of the iterations [here](https://github.com/gorilla/mux/blob/master/route.go#L52) `matchErr` is set `ErrMethodMismatch` and it will return false. I hope you understand what I'm trying to say here.

Reply:
Okay, `.Methods(methods ...string)` seems to act like a set and by chaining them makes them do some kind of intersection operation.

```go
.Methods("GET", "POST").Methods("PATCH") // will return 405 for all requests

.Methods("GET", "POST").Methods("GET", "PATCH") // will return 405 for all requests but GET requests
```